### DISCUSSION TITLE
What is the future of the CRD Validator component?

### DISCUSSION THREAD

**Discussion by tylerslaton** [operator-framework/rukpak#589] [Q&A]
## Summary
We've seen some value derived from the current webhook implementation of this component. However, there are some operational costs that we're not quite willing to pay that come along with it. One potential solution to this is instead doing this in-tree without any additional webhooks.

What are our thoughts here?

## Additional context
https://github.com/operator-framework/rukpak/issues/289
https://github.com/operator-framework/rukpak/issues/368

**Comment by exdx** [comment]
I think rukpak should be as unopinionated as possible when it comes to the content that it installs. I think the value-add of the validator webhook is marginal, and basically a copy/paste from OLM. In OLM we were highly opinionated on what we could install and upgrade, and I think that was a detriment to its growth. 

I would personally be for making a new repository for the CRD validator webhook code, and storing it there. Remove it from rukpak altogether. Then we could make it available via a Bundle, once rukpak supports webhooks.
